src/read.lisp

```diff
@@ -77,9 +77,9 @@
 (defun make-string-stream (string)
   (cons string 0))
 
-(defun %peek-char (stream)
-  (and (< (cdr stream) (length (car stream)))
-       (char (car stream) (cdr stream))))
+(defun %peek-char (stream &optional (look-ahead 0))
```

Comment:
I seem to recall this needed for guessing whether to dispatch `#\U+`*xx* or `#\`*name-char*. That bit can probably be rewritten to to use its own buffer to fetch the first couple of characters itself. I don't think I resorted to this elsewhere.

For near-future use-case, I'd expect the next sources of streams would rather be the results of an XHR call, arriving as a “whole” buffer at a time, so this seemed like a “safe enough” operation for that case. 

Admittedly there could be some streams in future that can't support a read-ahead at all, but I don't off-hand know of one that can accept a one-char-ahead peek and not accept a 2-char-ahead peek.